lib/types/json-schema.ts

```diff
@@ -53,13 +53,12 @@ export type JSONSchemaType<T, _partial extends boolean = false> = (T extends num
   : T extends Record<string, any>
   ? {
       // JSON AnySchema for records and dictionaries
-      // "required" is not optional because it is often forgotten
       // "properties" are optional for more concise dictionary schemas
       // "patternProperties" and can be only used with interfaces that have string index
       type: JSONType<"object", _partial>
       // "required" type does not guarantee that all required properties are listed
       // it only asserts that optional cannot be listed
-      required: _partial extends true ? Readonly<(keyof T)[]> : Readonly<RequiredMembers<T>[]>
+      required?: _partial extends true ? Readonly<(keyof T)[]> : Readonly<RequiredMembers<T>[]>
```

Comment:
A possible improvement is to make required optional if RequiredMembers is an empty array, if RequiredMembers then required must be used in the schema.